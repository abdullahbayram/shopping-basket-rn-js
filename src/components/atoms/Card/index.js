import { Card as PaperCard } from 'react-native-paper';

export default PaperCard;
