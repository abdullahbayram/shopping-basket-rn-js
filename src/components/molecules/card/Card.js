import * as React from 'react';
import { Button, Card as PaperCard, Text } from 'react-native-paper';
import PropTypes from 'prop-types';
import { StyleSheet } from 'react-native';

const renderRightContent = (price) => <Text variant="titleSmall">{price}</Text>;

const ProductCard = ({ buttonTitle, subtitle, title, price }) => (
  <PaperCard style={styles.container}>
    <PaperCard.Title style={styles.title} title={title} right={() => renderRightContent(price)} subtitle={subtitle} />
    <PaperCard.Actions>
      <Button>{buttonTitle}</Button>
    </PaperCard.Actions>
  </PaperCard>
);

export default ProductCard;

ProductCard.propTypes = {
  buttonTitle: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
};

const styles = StyleSheet.create({
  container: {
    marginVertical: 10,
  },
  title: {
    flexDirection: 'row',
    paddingHorizontal: 16,
  },
});
