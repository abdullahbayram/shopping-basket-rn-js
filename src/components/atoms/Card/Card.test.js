import React from 'react';
import { render } from '@testing-library/react-native';
import Card from '.';

test('Card renders correctly', () => {
  const view = render(<Card />);
  expect(view.toJSON()).toMatchSnapshot();
});
