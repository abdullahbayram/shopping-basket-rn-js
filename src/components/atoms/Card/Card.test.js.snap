// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
<View
  collapsable={false}
  style={
    {
      "backgroundColor": "rgba(247, 243, 249, 1)",
      "borderRadius": 12,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.15,
      "shadowRadius": 3,
    }
  }
  testID="card-container-outer-layer"
>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "rgba(247, 243, 249, 1)",
        "borderRadius": 12,
        "elevation": 1,
        "flex": undefined,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 1,
      }
    }
    testID="card-container"
  >
    <View
      style={
        [
          {
            "flexShrink": 1,
          },
          undefined,
        ]
      }
      testID="card"
    />
  </View>
</View>
`;
